tests/unit/library/test_json_stringformat.py

```diff
@@ -0,0 +1,941 @@
+            "null",  # all string formats ignore nulls
+            '"http://foo.bar/?baz=qux#quux"',  # a valid URI
+            '"//foo.bar/?baz=qux#quux"',  # a valid protocol-relative URI Reference
+            '"/abc"',  # a valid relative URI Reference
+            '"abc"',  # a valid URI Reference
+            '"#fragment"',  # a valid URI fragment
+        ],
+    )
+    def test_good(self, target_str):
+        schema_obj = json.loads(self.schema)
+        target_obj = json.loads(target_str)
+        generate_and_check(target_obj, schema_obj)
+
+    @pytest.mark.parametrize(
+        "bad_str",
+        [
+            '"\\\\\\\\WINDOWS\\\\fileshare"',  # an invalid URI Reference
+            '"#frag\\\\ment"',  # an invalid URI fragment
+        ],
+    )
+    def test_bad(self, bad_str):
+        schema_obj = json.loads(self.schema)
+        check_match_failure(bad_string=bad_str, schema_obj=schema_obj, compact=True)
+
+
+class TestTime:
+    schema = '{"$schema":"https://json-schema.org/draft/2020-12/schema","format":"time"}'
+
+    @pytest.mark.parametrize(
+        "target_str",
+        [
+            "12",  # all string formats ignore integers
+            "13.7",  # all string formats ignore floats
+            "{}",  # all string formats ignore objects
+            "[]",  # all string formats ignore arrays
+            "false",  # all string formats ignore booleans
+            "null",  # all string formats ignore nulls
+            '"08:30:06Z"',  # a valid time string
+            '"23:59:60Z"',  # a valid time string with leap second, Zulu
+            '"23:59:60+00:00"',  # valid leap second, zero time-offset
+            '"01:29:60+01:30"',  # valid leap second, positive time-offset
+            '"23:29:60+23:30"',  # valid leap second, large positive time-offset
+            '"15:59:60-08:00"',  # valid leap second, negative time-offset
+            '"00:29:60-23:30"',  # valid leap second, large negative time-offset
+            '"23:20:50.52Z"',  # a valid time string with second fraction
+            '"08:30:06.283185Z"',  # a valid time string with precise second fraction
+            '"08:30:06+00:20"',  # a valid time string with plus offset
+            '"08:30:06-08:00"',  # a valid time string with minus offset
+            '"08:30:06z"',  # a valid time string with case-insensitive Z
+        ],
+    )
+    def test_good(self, target_str):
+        schema_obj = json.loads(self.schema)
+        target_obj = json.loads(target_str)
+        generate_and_check(target_obj, schema_obj)
+
+    @pytest.mark.parametrize(
+        "bad_str",
+        [
+            '"008:030:006Z"',  # invalid time string with extra leading zeros
+            '"8:3:6Z"',  # invalid time string with no leading zero for single digit
+            '"8:0030:6Z"',  # hour, minute, second must be two digits
+            pytest.param('"22:59:60Z"', marks=pytest.mark.xfail(reason="leap seconds are hard")),  # invalid leap second, Zulu (wrong hour)
```

Comment:
Happy to see leap seconds acknowledged :-)

Reply:
I can't take credit 😉